pytest/lib/cluster.py

```diff
@@ -386,14 +382,13 @@ def resume_network(self):
 
 class GCloudNode(BaseNode):
 
-    def __init__(self, *args):
-        if len(args) == 1:
+    def __init__(self, instance_name=None, username=None, project=None, ssh_key_path=None, rpc_port=3030, *args):
+        if len(args) == 0:
             # Get existing instance assume it's ready to run
-            name = args[0]
-            self.instance_name = name
+            self.instance_name = instance_name
             self.port = 24567
             self.rpc_port = 3030
-            self.machine = gcloud.get(name)
+            self.machine = gcloud.get(instance_name, username=username, project=project, ssh_key_path=ssh_key_path)
             self.ip = self.machine.ip
         elif len(args) == 4:
```

Comment:
This branch looks weird now. Should this be `3` now? Should the
prototype of `__init__` be:

```
def __init__(self, instance_name=None, *args, username=None, project=None, ssh_key_path=None, rpc_port=3030):
    ...
```

I think that’s valid in Python 3.7 (which we seem to be targeting sinc
that’s what’s on NayDuck).